lib/orderbook/OrderBookRepository.ts

```diff
@@ -30,6 +31,15 @@ class OrderbookRepository {
   public addPairs = (pairs: db.PairFactory[]): Bluebird<db.PairInstance[]> => {
     return this.models.Pair.bulkCreate(<db.PairAttributes[]>pairs);
   }
+
+  public addOrderIfNotExists = async (order: db.OrderFactory) => {
```

Comment:
Yeah, I could not figure out a way to do this without resorting to raw SQL queries, and frankly I'm not sure how much more performant it is to do a `INSERT INTO... SELECT` vs this, but either way it's not a big deal because it's embedded like you said and it doesn't block the node process.